src/Orchard.Web/Modules/Orchard.OutputCache/Filters/OutputCacheFilter.cs

```diff
@@ -207,6 +207,15 @@ public class OutputCacheFilter : FilterProvider, IActionFilter, IResultFilter, I
                 var response = filterContext.HttpContext.Response;
                 var captureStream = new CaptureStream(response.Filter);
                 response.Filter = captureStream;
+
+                // Add ETag header for the newly created item
+                var etag = Guid.NewGuid().ToString("n");
+                if (HttpRuntime.UsingIntegratedPipeline) {
```

Comment:
Is there actually a case when this can be false? I guess only when running SpecFlow tests?